worker/multiwatcher/worker_internal_test.go

```diff
@@ -0,0 +1,361 @@
+		watcher: w1,
+		reply:   make(chan bool, 1),
+	}
+	sm.handle(req2)
+	assertWaitingRequests(c, sm, map[*Watcher][]*request{
+		w0: {req1, req0},
+		w1: {req2},
+	})
+
+	// Stop first watcher.
+	sm.handle(&request{
+		watcher: w0,
+	})
+	assertWaitingRequests(c, sm, map[*Watcher][]*request{
+		w1: {req2},
+	})
+	assertReplied(c, false, req0)
+	assertReplied(c, false, req1)
+
+	// Stop second watcher.
+	sm.handle(&request{
+		watcher: w1,
+	})
+	assertWaitingRequests(c, sm, nil)
+	assertReplied(c, false, req2)
+}
+
+func (*workerSuite) TestRespondMultiple(c *gc.C) {
+	sm := &Worker{
+		config: Config{
+			Logger:  loggo.GetLogger("test.worker"),
+			Backing: testbacking.New(nil),
+		},
+		request: make(chan *request),
+		waiting: make(map[*Watcher]*request),
+		store:   multiwatcher.NewStore(loggo.GetLogger("test.store")),
+	}
+
+	sm.store.Update(&multiwatcher.MachineInfo{ID: "0"})
+
+	// Add one request and respond.
+	// It should see the above change.
+	w0 := sm.newWatcher(nil)
+	req0 := &request{
+		watcher: w0,
+		reply:   make(chan bool, 1),
+	}
+	sm.handle(req0)
+	sm.respond()
+	assertReplied(c, true, req0)
+	c.Assert(req0.changes, gc.DeepEquals, []multiwatcher.Delta{{Entity: &multiwatcher.MachineInfo{ID: "0"}}})
+	assertWaitingRequests(c, sm, nil)
+
+	// Add another request from the same watcher and respond.
+	// It should have no reply because nothing has changed.
+	req0 = &request{
+		watcher: w0,
+		reply:   make(chan bool, 1),
+	}
+	sm.handle(req0)
+	sm.respond()
+	assertNotReplied(c, req0)
+
+	// Add two requests from another watcher and respond.
+	// The request from the first watcher should still not
+	// be replied to, but the later of the two requests from
+	// the second watcher should get a reply.
+	w1 := sm.newWatcher(nil)
+	req1 := &request{
+		watcher: w1,
+		reply:   make(chan bool, 1),
+	}
+	sm.handle(req1)
+	req2 := &request{
+		watcher: w1,
+		reply:   make(chan bool, 1),
+	}
+	sm.handle(req2)
+	assertWaitingRequests(c, sm, map[*Watcher][]*request{
+		w0: {req0},
+		w1: {req2, req1},
+	})
+	sm.respond()
+	assertNotReplied(c, req0)
+	assertNotReplied(c, req1)
+	assertReplied(c, true, req2)
+	c.Assert(req2.changes, gc.DeepEquals, []multiwatcher.Delta{{Entity: &multiwatcher.MachineInfo{ID: "0"}}})
+	assertWaitingRequests(c, sm, map[*Watcher][]*request{
+		w0: {req0},
+		w1: {req1},
+	})
+
+	// Check that nothing more gets responded to if we call respond again.
+	sm.respond()
+	assertNotReplied(c, req0)
+	assertNotReplied(c, req1)
+
+	// Now make a change and check that both waiting requests
+	// get serviced.
+	sm.store.Update(&multiwatcher.MachineInfo{ID: "1"})
+	sm.respond()
+	assertReplied(c, true, req0)
+	assertReplied(c, true, req1)
+	assertWaitingRequests(c, sm, nil)
+
+	deltas := []multiwatcher.Delta{{Entity: &multiwatcher.MachineInfo{ID: "1"}}}
+	c.Assert(req0.changes, gc.DeepEquals, deltas)
+	c.Assert(req1.changes, gc.DeepEquals, deltas)
+}
+
+var respondTestChanges = [...]func(store multiwatcher.Store){
```

Comment:
Would be good to test multi models. There is filtering logic for models, so unless it's tested elsewhere and I'm blind, would be good to test it  somewhere here.

Reply:
Will look and add if missing.

Reply:
watcher_test.go has tests for the multiple uuids.